ratatui-widgets/src/chart.rs

```diff
@@ -895,21 +928,45 @@ impl<'a> Chart<'a> {
             Alignment::Right => Alignment::Left,
         };
 
+        // The first label will be drawn in the space offered by y labels on the left
         Self::render_label(buf, labels.first().unwrap(), label_area, label_alignment);
 
+        // Finding positions for the labels by incrementing the
+        // `width_per_tick` each step is not a good solution because
+        // it accumulates rounding error at each iteration.
+        // A correct solution is to find the center position for the
+        // label box, i.e. where the tick mark should be, in terms
+        // of percentage of the graph area's width.
+        let width_percentage_increment = 1.0 / f64::from(labels_len - 1);
+
         for (i, label) in labels[1..labels.len() - 1].iter().enumerate() {
-            // We add 1 to x (and width-1 below) to leave at least one space before each
-            // intermediate labels
-            let x = graph_area.left() + (i + 1) as u16 * width_between_ticks + 1;
-            let label_area = Rect::new(x, y, width_between_ticks.saturating_sub(1), 1);
+            let current_label_width_percentage = (i + 1) as f64 * width_percentage_increment;
+
+            let x = linear_interpolation(
+                graph_area.left(),
+                graph_area.right(),
+                current_label_width_percentage,
+            );
+
+            ticks.push(x);
+
+            // We offset x by minus half the `width_per_tick` to find the left end
+            // of the label's bounding box.
+            let label_area = Rect::new(x - width_per_tick / 2, y, width_per_tick, 1);
 
             Self::render_label(buf, label, label_area, Alignment::Center);
         }
 
-        let x = graph_area.right() - width_between_ticks;
-        let label_area = Rect::new(x, y, width_between_ticks, 1);
+        // Leaving as much space as possible for the last tick, all while respecting
+        // the average tick bounding box proportions.
```

Comment:
This comment explains the result without the rationale. Why are we leaving this space?

Reply:
Before I amend the comment, let me explain and see whether we are aligned on this particular topic: sometimes the last tick might have more or less columns available that what was allocated per tick on the axis because of rounding variations. The ideal scenario is to leave as much space to the last tick as possible but there are 2 issues with this: 1. this requires knowing the rendered width of the penultimate tick (can be done using `.width()` of course, but do we want to rely on this information?) and 2. it means the last tick might have a different proportion than the others, which is maybe something we also want to avoid?